doc/users/next_whats_new/separated_hatchcolor.rst

```diff
@@ -0,0 +1,58 @@
+Separated ``hatchcolor`` from ``edgecolor``
+-------------------------------------------
+
+`~matplotlib.patches.Patch` gained a new *hatchcolor* parameter to explicitly
+control hatch colors. Previously, hatch colors were the same as edge colors,
+with a fallback to :rc:`hatch.color` if the patch did not have an edge color.
+
+Fallback Logic
+~~~~~~~~~~~~~~
+When the *hatchcolor* parameter is specified, it will be used for the hatch.
+If it is not specified, it will fallback to using :rc:`hatch.color`.
+
+The special value 'edge' takes over the patch edgecolor, with a fallback to
+:rc:`patch.edgecolor` if the patch edgecolor is 'none'.
+
+.. plot::
+    :include-source: true
+    :alt: Example of using the hatchcolor parameter in a Rectangle
+
+    import matplotlib as mpl
+    import matplotlib.pyplot as plt
+    from matplotlib.patches import Rectangle
+
+    fig, ax = plt.subplots()
+
+    # hatchcolor can now be controlled using the `hatchcolor` parameter
+    # In this case, hatchcolor is orange
+    patch1 = Rectangle((0.1, 0.1), 0.3, 0.3, edgecolor='red', linewidth=2,
+                      hatch='//', hatchcolor='orange')
+    ax.add_patch(patch1)
+
+    # When hatchcolor is not specified, it matches edgecolor
+    # In this case, hatchcolor is green
+    patch2 = Rectangle((0.6, 0.1), 0.3, 0.3, edgecolor='green', linewidth=2,
+                       hatch='//', facecolor='none')
+    assert patch2._hatch_color == 'edge'
+    assert patch2.get_hatchcolor() == mpl.colors.to_rgba('green')
```

Comment:
Do we really need asserts for the example?

Reply:
Yup, asserts seem unnecessary

Reply:
I've mixed feelings: Typically not. Here it's a way to demonstrate that the default hatchcolor is "edge" and that the resulting color will thus be "green". But the comment above should be sufficient.

Reply:
Yes, the comments should be sufficient